### PR TITLE
[SMP] Modified to run two linuxs run on SMP

### PR DESCRIPTION
**Pull request by casionwoo**

### CODE REVIEW

File: arch/arm/arch_regs.c [hunk 1]
```diff
@@ -201,16 +201,20 @@ hvmm_status_t arch_regs_save(struct arch_regs *regs, struct core_regs *current_r
 extern void __set_vcpu_context_first_time(struct core_regs *regs);
 hvmm_status_t arch_regs_restore(struct arch_regs *regs, struct core_regs *current_regs)
 {
+
```
Comment (Wonseok) [inline arch/arm/arch_regs.c]: Remove the empty line.

File: arch/arm/arch_regs.c [hunk 2]
```diff
@@ -201,16 +201,20 @@ hvmm_status_t arch_regs_save(struct arch_regs *regs, struct core_regs *current_r
...
     // 'current_regs == 0' means there is no vcpu are running on hypervisor.
     // We need a way to boot first vcpu up as below.
     if (!current_regs) {
+        restore_cp15(&regs->cp15);
         __set_vcpu_context_first_time(&regs->core_regs);
+
+        /* Reach not here */
```
Comment (Wonseok) [inline arch/arm/arch_regs.c]: s/Reach not here/Never reach here/

File: core/sched/scheduler.c [hunk 3]
```diff
@@ -132,7 +127,12 @@ void sched_start(void)
 
     /* Select the first guest context to switch to. */
     uint64_t expiration = 0;
-    vcpu = vcpu_find(__policy[pcpu]->do_schedule(&expiration));
+    uint32_t vcpuid = VCPUID_INVALID;
+
+    /* NOTE : Hanging on while if there's no vcpu to schedule on scheduler */
+    while( (vcpuid = __policy[pcpu]->do_schedule(&expiration)) == VCPUID_INVALID) ;
```
Comment (Wonseok) [inline core/sched/scheduler.c]: It's too long and looks odd. Change it into three lines as below:
vcpuid = __policy[pcpu]->do_schedule(&expiration);
while(vcpuid == VCPUID_INVALID) 
  ;

File: drivers/vdev/vdev_sysreg.c [hunk 4]
```diff
@@ -46,7 +48,7 @@ struct vsysreg {
     uint32_t sys_sw;
     uint32_t sys_led;
     uint32_t sys_100hz;
-    uint32_t sys_flag;
+    uint32_t sys_flags;
     uint32_t sys_flagsset;
```
Comment (Wonseok) [inline drivers/vdev/vdev_sysreg.c]: Removed unused variables in structure.

**Comment by Wonseok** [comment]
Overall looks good. 
Please see the comments.

File: core/sched/scheduler.c (hunk 3 above)
Comment (casionwoo) [inline core/sched/scheduler.c]: We should call do_schedule until return is not 'VCPUID_INVALID' for checking if there's vcpu to schedule.

**Pull request merged**